src/FParsec.CSharp/CharParsersCS.cs

```diff
@@ -1404,6 +1404,29 @@ public static class CharParsersCS {
         /// <summary>`UByteU()` behaves like `UByte`, but supports user state.</summary>
         public static FSharpFunc<CharStream<U>, Reply<byte>> UByteU<U>() => puint8<U>();
 
+        /// <summary>
+        /// <para>
+        /// Parses a number literal and returns the result in form of a `NumberLiteral` value.
+        /// </para>
+        /// <para>
+        /// The given `NumberLiteralOptions` argument determines the kind of number literals accepted.
+        /// The string `label` is used in the `Expected` error message that is generated when the parser fails without consuming input. 
+        /// </para>
+        /// <para>
+        /// The parser fails without consuming input, if not at least one digit (including the 0 in the format specifiers "0x" etc.) can be parsed.
+        /// It fails after consuming input, if no decimal digit comes after an exponent marker or no valid digit comes after a format specifier.
+        /// </para>
+        /// </summary>
+        public static FSharpFunc<CharStream<U>, Reply<NumberLiteral>> NumberLiteral<U>(NumberLiteralOptions opt, string label) => numberLiteral<U>(opt, label);

```

Comment:
Have two variants with and without userstate:

```suggestion
        public static FSharpFunc<CharStream<Unit>, Reply<NumberLiteral>> NumberLiteral(NumberLiteralOptions opt, string label) => numberLiteral<Unit>(opt, label);

        /// <summary>`NumberLiteralU()` behaves like `NumberLiteral()`, but supports user state.</summary>
        public static FSharpFunc<CharStream<U>, Reply<NumberLiteral>> NumberLiteralU<U>(NumberLiteralOptions opt, string label) => numberLiteral<U>(opt, label);
```